Merge Game Origin Rewrite branch and master

jnackmclain opened the pull request:
This PR merges all upstream into game origin rewrite. I spent a great deal of time doing my best to keep the newer data labeling in-tact where applicable. I did my best but keep in mind I am only just learning. It would be wise to make sure and give this a nice look over to ensure accuracy. I did spend my time on this, I did not just throw this in chatgpt.

(sorry for commit spam)

What is not done:
Relevant Wii Bank 8 port addresses, I only merged what existed, i did not fetch those.
everything else comp mentioned, this only merges upstream.

Tested:
Builds Wii via github actions, builds Xbox locally
Tested locally on Xbox 360 hardware, everything seems correct
new icon rewrite works, double bass pedal modifier works, dtafunctions work, gocentral works, song id correction works, everything appears in order.

Not Tested:
Actually running on wii (my workflow is very heavily focused on bank8)

InvoxiPlayGames committed:
try to fix bug when missing song data specifies song id

InvoxiPlayGames committed:
Merge pull request #18 from RBEnhanced/song_id_fixer_fix

Try to fix an issue when song_id is specified in missing song data

InvoxiPlayGames committed:
temporarily disable wii usb hooks until i get it working

InvoxiPlayGames committed:
File vtable

ihatecompvir committed:
Patch to fix reading version 34 meshes

ihatecompvir committed:
Merge branch 'master' of https://github.com/RBEnhanced/RB3Enhanced

ihatecompvir committed:
Fix Wii build + note shuffle

ihatecompvir committed:
Support 2x bass pedal on MIDI note 95

allows 2x bass pedal versions of songs to exist in one MIDI. plus there is a new modifier to toggle this

ihatecompvir committed:
Unlock all clothing no longer shows lock icon

ihatecompvir committed:
Fix double bass

Forgot that I need to actually call the original function, lol

Blasteroids committed:
SK Led fix

Fixes the RB3 (bug?) where 2 red leds stay on after exiting the score screen.  This fix turns them off.

InvoxiPlayGames committed:
Merge pull request #21 from Blasteroids/master

SK Led fix on score overview exit

InvoxiPlayGames committed:
fixes a crash when playing online

InvoxiPlayGames committed:
Merge branch 'master' of https://github.com/RBEnhanced/RB3Enhanced

DragRedSim committed:
Extend strncpy(config.GoCentralAddress) to full field length

config.GoCentralAddress is defined in the ini spec in config.h as being of up to RB3E_MAX_DOMAIN chars, but the strncpy routine for that item was only copying RB3E_MAX_CONFIG_LEN; this was 30 chars. The announced Xbox URI had precisely 31 characters. This caused all DNS lookups for this address to fail, and no connection to be made to the GoCentral server.

InvoxiPlayGames committed:
Merge pull request #24 from DragRedSim/patch-dns-name-length

Extend strncpy(config.GoCentralAddress) to full field length

jnackmclain committed:
add index created by @InvoxiPlayGames

adding the original version created by emma for tracking purposes before committing changes

InvoxiPlayGames committed:
Merge pull request #25 from jnackmclain/add_index

add index created by @InvoxiPlayGames

jnackmclain committed:
massively overhaul index.html

This updates the styling of the page to be a bit more modern. The page also has a theme per the running device state of light mode vs dark mode. Add's the ability to use (or hide) a lastfm api key for album art fetching. This isnt a perfect mechanism so you are also able to hide the api key field to not attempt getting album images. The album images are tagged and cached based on the corresponding song tag, so any arbitrary filtering of the song list will continue to match the cached image. filtered searches trigger faster api calls so the user isnt sat looking at a blank image. once search is unfiltered, the task for populating art for the entire library continues running. No longer need to click a search button to trigger a search, can live type and live filter.

InvoxiPlayGames committed:
rb3.ini in assets folder, fix actions..?

ihatecompvir committed:
Merge pull request #26 from jnackmclain/enhanced_index

massively overhaul rb3e_index.html

InvoxiPlayGames committed:
use my fork of brainslug in actions

InvoxiPlayGames committed:
remove compiler "errors" that should be warnings if not for the lovely toolchain update

ihatecompvir committed:
Don't trigger breakpoints on server errors

LinosM committed:
Support for scripts over the network

InvoxiPlayGames committed:
Merge pull request #27 from LinosM/master

Support for scripts over the network

InvoxiPlayGames committed:
add some scripting error debugging similar to debug

InvoxiPlayGames committed:
enable 192MB heap on Dolphin, groundwork for launcher to set it on vWii

InvoxiPlayGames committed:
use ubuntu-latest in Wii actions

InvoxiPlayGames committed:
bank 8 support, ports header restructure

InvoxiPlayGames committed:
more bank8 addy ports

InvoxiPlayGames committed:
rebuild DataRegisterFunc, fix keyboard on 256mb, build bank8 in actions

InvoxiPlayGames committed:
480p fix, deflicker fix (disabled by default, needs safety research)

InvoxiPlayGames committed:
use correct packet size for PCP (fix #32)

InvoxiPlayGames committed:
fix 360 DTA functions

InvoxiPlayGames committed:
Merge pull request #31 from RBEnhanced/bank8

Support Wii Bank 8, restructured ports.h files

ihatecompvir committed:
Xbox 360 socket timeout typo fix

Don't really think this was really causing any specific problems, but fixing anyway

InvoxiPlayGames committed:
[bank8] fix GetSongIDFromShortName to stop crash initialising TheGame

InvoxiPlayGames committed:
[bank8] use gSongMgr instead of TheSongMgr address

InvoxiPlayGames committed:
[wii] graphical exception handler for diagnosing crashes

InvoxiPlayGames committed:
[bank8] log fatal errors and dont restart after crashes

InvoxiPlayGames committed:
[wii] be slightly more mindful of CPU cache

InvoxiPlayGames committed:
[wii] FileSD backend for SD cards

InvoxiPlayGames committed:
Merge pull request #34 from RBEnhanced/wii-files

[Wii] New FileSD backend for SD cards

InvoxiPlayGames committed:
add DTA functions for getting song metadata by ID

InvoxiPlayGames committed:
[dolphin] allow read-only reads from NAND (for dx ace.dta)

InvoxiPlayGames committed:
[wii] fix dta interop

jnackmclain committed:
fix typo in get album

InvoxiPlayGames committed:
Merge pull request #36 from jnackmclain/master

fix typo in get album

jnackmclain committed:
A second typo has hit the album func (#37)

* wait is this it

* please run my action thanks

* preesh

* Update DTAFunctions.c

InvoxiPlayGames committed:
[wii] load encrypted SD DLC as if they were on NAND

InvoxiPlayGames committed:
[wii] correct filesize when enumerating CNTSD DLC, read key from SD

InvoxiPlayGames committed:
[wii] allow unencrypted .app DLC from SD, very small speed improvement

jnackmclain committed:
xbox: fix typo in song metadata struct (#41)

InvoxiPlayGames committed:
[wii] INT_VALUE = 6 on Wii

InvoxiPlayGames committed:
Merge branch 'master' of https://github.com/RBEnhanced/RB3Enhanced

jnackmclain committed:
Merge remote-tracking branch 'upstream/master' into game_origin_update

jnackmclain committed:
move ports to their proper spot

jnackmclain committed:
update calls to metadata struct

jnackmclain committed:
oh

jnackmclain committed:
oh duh

jnackmclain committed:
names may have shifted

jnackmclain committed:
these too

jnackmclain committed:
sure

jnackmclain committed:
well yeah of course

jnackmclain committed:
im bad at merging i guess

jnackmclain committed:
merge

jnackmclain committed:
Update ChunkStream.h

jnackmclain committed:
lawl

jnackmclain committed:
yeah bro

jnackmclain committed:
yeah lets try it sure

jnackmclain committed:
Update MiloSceneHooks.h

jnackmclain committed:
lol

jnackmclain committed:
ya

jnackmclain committed:
hail mary

jnackmclain committed:
idk

jnackmclain committed:
please

jnackmclain committed:
yeah

jnackmclain committed:
Update FileStream.h

jnackmclain committed:
add ports

jnackmclain committed:
unused?

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
bro this shit is ass

jnackmclain committed:
oof

jnackmclain committed:
idk

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
sure

jnackmclain committed:
Update SortNode.h

jnackmclain committed:
Update SortNode.h

jnackmclain committed:
Update SortNode.h

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
update calls

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
oh remove this?

jnackmclain committed:
Update NodeSort.h

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
list slot

jnackmclain committed:
list int

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
include mem

jnackmclain committed:
Update SetlistHooks.c

jnackmclain committed:
NodeSort

jnackmclain committed:
Update SongParserHooks.c

jnackmclain committed:
add vertex read hook

jnackmclain committed:
Update rb3enhanced.c

jnackmclain committed:
correct some issues i noticed

I put the wrong version of songhooks.c

jnackmclain committed:
Merge pull request #2 from jnackmclain/tidy_update

correct some issues i noticed

jnackmclain commented:
I did a full sanity check looking over my [mock PR](https://github.com/jnackmclain/RB3Enhanced/pull/1/files) files changed tab, and noticed i made a couple small errors, im otherwise feeling good about it.

InvoxiPlayGames committed:
[wii] add ModernSDMode toggle for emulated CNTSD

InvoxiPlayGames committed:
Merge pull request #38 from RBEnhanced/cntsd-simulation

[Wii] Load encrypted SD DLC as if they were on NAND

InvoxiPlayGames committed:
[chore] update building, contributing and readme

jnackmclain committed:
Merge remote-tracking branch 'upstream/master' into game_origin_update

jnackmclain commented:
added latest pushes from upstream

jnackmclain committed:
add bank8 ports

jnackmclain committed:
do not define this outside of bank8

jnackmclain committed:
wait this isnt even used

jnackmclain committed:
fix address

jnackmclain committed:
Merge pull request #3 from jnackmclain/i_really_tried

Add bank8 addresses, fix bank8 compile

jnackmclain committed:
bank8: fix enumerate fail when legacy SD mode is disabled (#44)

jnackmclain commented:
I got the milo highlight issue figured out for the xbox milo, unfortunately I am having issues with the wii milo

The fix for xbox was just to reorder the mesh in the primary.grp to the top, but it appears some differences in the wii mesh (different geom_owner and mat) make this a bit different. When I tried to duplicate the same mesh, the location of the icon moved to the right hand side. Might need a bit more attention from comp on the specific wii differences as I am not as versed in 010 to dig deeper

[list_song_select_browser.milo_xbox.zip](https://github.com/user-attachments/files/20474570/list_song_select_browser.milo_xbox.zip)

jnackmclain committed:
[feature] allow ini setting to disable metamusic loading (#45)

InvoxiPlayGames committed:
[xbox] graphical exception handler and exception dumper

jnackmclain committed:
Merge remote-tracking branch 'upstream/master' into game_origin_update

jnackmclain commented:
![image](https://github.com/user-attachments/assets/493bf8d5-c46f-46eb-a29e-889886c0bd16)

InvoxiPlayGames committed:
[feature] clear song cache from DTA, log content name/mount

Sulfrix commented:
Edited the song select list milo to move the highlight mesh left to accommodate the source icon. I also adjusted the positioning of the icon very slightly. This (imo) makes the icon fit in better than *just* proper ordering.
[list_song_select_browser.milo_xbox.zip](https://github.com/user-attachments/files/21675228/list_song_select_browser.milo_xbox.zip)
<img width="1280" height="720" alt="icons" src="https://github.com/user-attachments/assets/65b8638c-b927-4857-bcdc-221113ce0ab4" />
No Wii milo edit sadly, I don't have experience with editing those.

InvoxiPlayGames committed:
[chore] housekeeping

Sulfrix committed:
[xbox] add keyboard support back into xbox retail (#49)

* [xbox] add keyboard support back into xbox retail

* clean up TranslateVK

jnackmclain committed:
[http] fix only 1 search result with no api key (#50)

jnackmclain committed:
Merge remote-tracking branch 'upstream/master' into game_origin_update

jnackmclain committed:
Merge remote-tracking branch 'upstream/game-origin-rewrite' into game_origin_update

jnackmclain committed:
split ports to proper place

jnackmclain committed:
bank 8 ports

jnackmclain committed:
song_select rawfile

jnackmclain committed:
keep alt style for artist

jnackmclain committed:
luna's updated milo

Co-Authored-By: luna <37005649+lunalawl@users.noreply.github.com>
Co-Authored-By: Sulfrix <22404294+Sulfrix@users.noreply.github.com>

jnackmclain committed:
attempt to fix song metadata struct genre offset (#46)

InvoxiPlayGames committed:
[dta] add several DTA functions and RB3E_HAS_VERSION define
rb3e_api_version, rb3e_build_tag, rb3e_commit and rb3e_local_ip

InvoxiPlayGames committed:
[bugfix] bixby, fix the wii compiler error

jnackmclain committed:
Merge remote-tracking branch 'upstream/master' into game_origin_update

InvoxiPlayGames committed:
[fix] remove buggy language hook, update index to refer to rbe webspace

ihatecompvir committed:
Load custom head milo when prefab head is hidden

this only impacts custom characters being loaded through modified prefabs and is purely to better support mods, there is no impact to vanilla rb3

ihatecompvir committed:
[bugfix] Check that the custom head milo exists before trying to load it

ihatecompvir committed:
[feature] INI setting to enable earning gold stars on non-Expert

ihatecompvir committed:
[feature] Dynamic character loader

automatically generates prefabs in-game for every character milo in /char/main/custom/characters/gen/

InvoxiPlayGames committed:
[bank8] fix actions builds

ihatecompvir committed:
[feature] Disable texture compression under Xenia

Allows the game to render character textures properly when not using readback resolve

InvoxiPlayGames committed:
[chore] fix formatting on xbox_keyboard.c

InvoxiPlayGames committed:
Merge branch 'master' of https://github.com/RBEnhanced/RB3Enhanced

jnackmclain committed:
refresh build actions

jnackmclain committed:
merge in upstream

jnackmclain committed:
Update rb3enhanced.c

jnackmclain committed:
fixes
